src/main/java/io/cdap/plugin/mainframe/config/MainframeRecordReaderConfig.java

```
@@ -0,0 +1,541 @@
+    super(referenceName);
+    this.copybook = copybook;
+    this.charset = charset;
+    this.inputPath = inputPath;
+    this.fileOrganization = fileOrganization;
+    this.dialect = dialect;
+    this.copybookSplit = copybookSplit;
+    this.recordDeciderFieldName = recordDeciderFieldName;
+    this.recordDeciderConditions = recordDeciderConditions;
+    this.recordSelectorFieldMappings = recordSelectorFieldMappings;
+    this.recordSelectorExpression = recordSelectorExpression;
+    this.recordSelectorUnmatchedToError = recordSelectorUnmatchedToError;
+    this.recordsPerPartition = recordsPerPartition;
+  }
+
+  /**
+   * @return a string representation of copybook.
+   */
+  public String getCopybook() {
+    return copybook;
+  }
+
+  /**
+   * @return true if copybook is a macro, false otherwise.
+   */
+  public boolean isCopybookMacro() {
+    return containsMacro(PROPERTY_COPYBOOK);
+  }
+
+  /**
+   * @return Encoding to be used for the parsing the binary file.
+   */
+  public String getCharset() {
+    return charset == null || !charsetMappings.containsKey(charset) ? "cp037" : charsetMappings.get(charset);
+  }
+
+  /**
+   * @return true if charset is a macro, false otherwise.
+   */
+  public boolean isCharsetMacro() {
+    return containsMacro(PROPERTY_CHARSET);
+  }
+
+  /**
+   * @return input path or file that needs to be processed.
+   */
+  public String getInputPath() {
+    return inputPath;
+  }
+
+  /**
+   * @return true if input path is a macro, false otherwise.
+   */
+  public boolean isInputPathMacro() {
+    return containsMacro(PROPERTY_INPUT_PATH);
+  }
+
+  /**
+   * @return the structure of the file being processed.
+   */
+  public int getFileOrganization() {
+    if (fileOrganization != null && !fileOrganization.trim().isEmpty()) {
+      if (fileOrganizationMaps.containsKey(fileOrganization)) {
+        return fileOrganizationMaps.get(fileOrganization);
+      }
+    }
+    return Constants.IO_FIXED_LENGTH;
```

if the user specified file organization doesn't exist, shouldn't that be a validation error instead of silently using something else?

---

If they don't specify anything default should be RECFM=F (IO_FIXED_LENGTH).